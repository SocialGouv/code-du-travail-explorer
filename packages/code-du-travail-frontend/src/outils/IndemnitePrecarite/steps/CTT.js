import React from "react";
import { Field } from "react-final-form";
import { Toast } from "@socialgouv/react-ui";

import { YesNoQuestion } from "../../IndemniteLicenciement/components/YesNoQuestion";

function validate(values) {
  const errors = {};
  if (values.cttFormation === true) {
    errors.cttFormation =
      "Votre type de contrat ne vous permet pas d’avoir droit à une prime de précarité.";
  }
  if (values.ruptureContratFauteGrave === true) {
    errors.ruptureContratFauteGrave =
      "Lorsque le contrat de travail temporaire (contrat d'intérim) est rompu de manière anticipée à l’initiative du salarié, pour faute grave ou en cas de force majeure, le salarié n’a pas le droit à une prime de précarité.";
  }
  if (values.propositionCDIFinContrat === true) {
    errors.propositionCDIFinContrat =
      "Le salarié en contrat de travail temporaire (contrat d'intérim) qui est immédiatement embauché en CDI au sein de l'entreprise dans laquelle il effectuait sa mission n’a pas le droit à une prime de précarité.";
  }
  if (values.refusSouplesse === true) {
    errors.refusSouplesse =
      "Le salarié en contrat d'intérim qui refuse la mise en œuvre de la souplesse prévue dans son contrat n’a pas le droit à une prime de précarité.";
  }
  return errors;
}

function StepCTT() {
  return (
    <>
      <YesNoQuestion
        label="Avez vous un contrat de mission-formation&nbsp;?"
        name="cttFormation"
        data-testid="cttFormation"
      />
      <Field name="cttFormation">
        {({ input }) =>
          input.value === false ? (
            <Toast variant="info">
              Attention : si vous avez un contrat de travail temporaire
              saisonnier ou d’usage, un accord d’entreprise ou d’établissement
              peut dispenser votre entreprise de travail temporaire
              (l’entreprise d’intérim) de vous verser une prime de précarité.
            </Toast>
          ) : null
        }
      </Field>
      <YesNoQuestion
        data-testid="ruptureContratFauteGrave"
        name="ruptureContratFauteGrave"
        label="Votre contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes : votre propre initiative, votre faute grave ou en cas de force majeure&nbsp;?"
      />
      <YesNoQuestion
        data-testid="propositionCDIFinContrat"
        name="propositionCDIFinContrat"
        label="À la fin de votre contrat d'intérim, avez-vous été immédiatement embauché en CDI au sein de l'entreprise dans laquelle vous effectuiez votre mission&nbsp;?"
      />
      <YesNoQuestion
        data-testid="refusSouplesse"
        name="refusSouplesse"
        label="Avez-vous refusé la mise en œuvre de la souplesse prévue dans votre contrat d’intérim&nbsp;?"
      />
    </>
  );
}

StepCTT.validate = validate;

export { StepCTT };
