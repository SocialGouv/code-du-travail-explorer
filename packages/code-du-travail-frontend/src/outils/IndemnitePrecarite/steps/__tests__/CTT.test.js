import React from "react";
import { render, fireEvent } from "react-testing-library";
import { StepCTT } from "../CTT";
import { Form } from "react-final-form";

function renderForm() {
  return render(
    <Form validate={StepCTT.validate} onSubmit={jest.fn()}>
      {({ form }) => <StepCTT form={form} />}
    </Form>
  );
}

const testQuestions = [
  ["ruptureContratFauteGrave", true],
  ["propositionCDIFinContrat", true],
  ["refusSouplesse", true]
];

describe("<StepCTT />", () => {
  it("should render", () => {
    const { container } = renderForm();
    expect(container).toMatchSnapshot();
  });

  test.each(testQuestions)(
    "should render error when %s is %s",
    (key, value) => {
      const { getByText, getByTestId } = renderForm();
      const radio = getByTestId(key).querySelector(`input[value=${value}]`);
      fireEvent.click(radio);
      // blur is need to force validation in react-testing-lib
      fireEvent.blur(radio);
      expect(getByText(/pas le droit Ã  une prime/i)).toBeTruthy();
    }
  );
});
