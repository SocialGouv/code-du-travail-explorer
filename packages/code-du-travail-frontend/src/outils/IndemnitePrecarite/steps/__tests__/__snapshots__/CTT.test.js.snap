// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepCTT /> should render 1`] = `
.c2 {
  padding: 0;
  font-size: 1.15rem;
  margin-right: 2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1.25rem;
}

.c0 {
  font-size: 1.25rem;
  display: block;
  margin-top: 1.25rem;
  margin-bottom: 0.625rem;
}

<div>
  <p
    class="c0"
  >
    Avez vous un contrat de mission-formation ?
  </p>
  <div
    class="c1"
    data-testid="cttFormation"
  >
    <label
      class="c2"
    >
      <input
        name="cttFormation"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="cttFormation"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c0"
  >
    Votre contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes : votre propre initiative, votre faute grave ou en cas de force majeure ?
  </p>
  <div
    class="c1"
    data-testid="ruptureContratFauteGrave"
  >
    <label
      class="c2"
    >
      <input
        name="ruptureContratFauteGrave"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="ruptureContratFauteGrave"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c0"
  >
    À la fin de votre contrat d'intérim, avez-vous été immédiatement embauché en CDI au sein de l'entreprise dans laquelle vous effectuiez votre mission ?
  </p>
  <div
    class="c1"
    data-testid="propositionCDIFinContrat"
  >
    <label
      class="c2"
    >
      <input
        name="propositionCDIFinContrat"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="propositionCDIFinContrat"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c0"
  >
    Avez-vous refusé la mise en œuvre de la souplesse prévue dans votre contrat d’intérim ?
  </p>
  <div
    class="c1"
    data-testid="refusSouplesse"
  >
    <label
      class="c2"
    >
      <input
        name="refusSouplesse"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="refusSouplesse"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;
